src/main/python/main.py

```diff
@@ -0,0 +1,1123 @@
+        raise NotImplementedError
+
+    def visit_Try(self, node: ast.Try):
+        debug_print(ast.dump(node))
+        # TODO parse body of try statement
+        t = TryStatement()
+
+        # node.body
+
+        return t
+
+    def visit_ExceptHandler(self, node):
+        debug_print(ast.dump(node))
+        raise NotImplementedError
+
+    def visit_With(self, node):
+        debug_print(ast.dump(node))
+        return  # TODO LATER -> new cpg node or foreach + try/catch?
+        raise NotImplementedError
+
+    def visit_withitem(self, node):
+        debug_print(ast.dump(node))
+        raise NotImplementedError
+
+    ### FUNCTION AND CLASS DEFINITIONS ###
+    def visit_FunctionDef(self, node: ast.FunctionDef, returnMethod=False):
+        debug_print(ast.dump(node))
+        if returnMethod:
+            fd = MethodDeclaration()
+        else:
+            fd = FunctionDeclaration()
+        self.add_loc_info(node, fd)
+
+        # handle name
+        fd.setName(node.name)
+        self.scopemanager.enterScope(fd)
+
+        annotations = ArrayList()
+
+        for decorator in node.decorator_list:
+            # cannot do this because kw arguments are not properly handled yet in functions
+            # expr = self.visit(decorator)
+
+            members = ArrayList()
+            annotation = Annotation()
+            self.add_loc_info(decorator, annotation)
+
+            if isinstance(decorator.func, ast.Attribute):
+                ref = self.visit(decorator.func)
+
+                annotation.setName(ref.getName())
+
+                # add the base as a receiver annotation
+                member = AnnotationMember()
+                self.add_loc_info(decorator.func, member)
+                member.setName("receiver")
+                member.setValue(ref.getBase())
+
+                members.add(member)
+            elif isinstance(decorator, ast.Name):
+                ref = self.visit(decorator.func)
+
+                annotation.setName(ref.getName())
+
+            # add first arg as value
+            if len(decorator.args) > 0:
+                arg0 = decorator.args[0]
+                value = self.visit(arg0)
+
+                member = AnnotationMember()
+                self.add_loc_info(arg0, value)
+                member.setName("value")
+                member.setValue(value)
+
+                members.add(member)
+
+            # loop through keywords args
+            for kw in decorator.keywords:
+                member = AnnotationMember()
+                member.setName(kw.arg)
+                member.setValue(self.visit(kw.value))
+
+                members.add(member)
+
+            annotation.setMembers(members)
+            annotations.add(annotation)
+
+        fd.addAnnotations(annotations)
+
+        # debug_print(node.de)
+
+        # handle args
+        # scopeManager adds them to fd
+        self.visit_arguments(node.args, fd)
+
+        # handle body
+        body = CompoundStatement()
```

Comment:
Use `make_compound_statement`

Reply:
Resolved in 5ec26c85f37b9d3fda77d70d502110e9e53196a4